src/example-types/metricGroups/pivot.js

```diff
@@ -321,7 +321,25 @@ let createPivotScope = (node, schema, getStats) => {
           group.sort = { field: sortField, direction: sort.direction }
         }
         let build = lookupTypeProp(_.identity, 'buildGroupQuery', group.type)
-        return build(group, await children, groupingType, schema, getStats)
+
+        //Remove anything that needs to be hoisted before composing further
+        let hoist = { hoistProps: { ...children.hoistProps } }
+        children = _.omit('hoistProps', await children)
+
+        let parent = await build(
+          group,
+          children,
+          groupingType,
+          schema,
+          getStats
+        )
+
+        //Add anything that needs to be hoisted to parent
+        parent.hoistProps = _.merge(
+          _.getOr({}, 'hoistProps', hoist),
+          parent.hoistProps
+        )
+        return parent
```

Comment:
Instead of juggling the hoisting stuff by stripping things out, it would be cleaner to just add a new optional method next to buildGroupQuery which you can call separately. You can avoid stripping and reconstructing stuff and also enable other use cases to enhance parents in a clear place

Reply:
I will push something to address the immediate concern of removing federal, then work on this refactoring work.